battle-engine.js

```diff
@@ -1256,6 +1256,23 @@ BattlePokemon = (function () {
 		if (this.battle.gen >= 5) return ['Normal'];
 		return ['???'];
 	};
+	BattlePokemon.prototype.isGrounded = function () {
+		if (!this.hasType('Flying') && !this.hasAbility('levitate') && !this.hasItem('airballoon') && !this.volatiles['magnetrise'] && !this.volatiles['telekinesis']) {
```

Comment:
Unless... are you saying this line should be `if (this.runImmunity('Ground', false) && !this.hasType('Flying')` instead?